exporters/readers/s3_reader.py

```diff
@@ -228,88 +187,11 @@ def __init__(self, *args, **kwargs):
                                                 self.read_option('aws_access_key_id'),
                                                 self.read_option('aws_secret_access_key'))
         self.keys = self.keys_fetcher.pending_keys()
-        self.read_keys = []
-        self.current_key = None
-        self.last_block = 0
         self.logger.info('S3Reader has been initiated')
-        self.tmp_folder = tempfile.mkdtemp()
-        self.lines_reader = self.read_lines_from_keys()
 
     def get_read_streams(self):
         from exporters.bypasses.stream_bypass import Stream
         for key_name in self.keys:
             key = self.bucket.get_key(key_name)
             file_obj = urlopen(key.generate_url(S3_URL_EXPIRES_IN))
             yield Stream(file_obj, key_name, key.size)
```

Comment:
This suffers from the same problem I mention in issue #322, except it behaves differently: instead of skipping the rest of the file and starting from next key, it just breaks (stacktrace below).

This is a better behavior than failing silently, so I'm not holding it against this PR, but the problem is still there.

```
INFO:export-pipeline: -- WRITER -- Checksum for file /tmp/tmpu7oh7r/fef43313-c007-45fd-b0f1-219cf6f34686/0000.jl.gz: 780e8b24b614f8cdcf761ed92267a198
ERROR:export-pipeline: -- EXPORTMANAGER -- Traceback (most recent call last):
  File "/home/elias/hacking/scrapinghub/exporters/exporters/export_managers/base_exporter.py", line 171, in export
    self._run_pipeline()
  File "/home/elias/hacking/scrapinghub/exporters/exporters/export_managers/base_exporter.py", line 161, in _run_pipeline
    self._run_pipeline_iteration()
  File "/home/elias/hacking/scrapinghub/exporters/exporters/export_managers/base_exporter.py", line 71, in _run_pipeline_iteration
    self.writer.write_batch(batch=next_batch)
  File "/home/elias/hacking/scrapinghub/exporters/exporters/writers/base_writer.py", line 90, in write_batch
    for item in batch:
  File "/home/elias/hacking/scrapinghub/exporters/exporters/readers/base_stream_reader.py", line 59, in get_next_batch
    yield next(self.iterator)
  File "/home/elias/hacking/scrapinghub/exporters/exporters/readers/base_stream_reader.py", line 41, in iteritems
    for record in items_iter:
  File "/home/elias/hacking/scrapinghub/exporters/exporters/deserializers.py", line 16, in deserialize
    for line in stream.iterlines():
  File "/home/elias/hacking/scrapinghub/exporters/exporters/iterio.py", line 132, in iterlines
    line = self.readline()
  File "/home/elias/hacking/scrapinghub/exporters/exporters/iterio.py", line 118, in readline
    line += self.next_chunk()
  File "/home/elias/hacking/scrapinghub/exporters/exporters/iterio.py", line 78, in next_chunk
    data = self._iterator.next()  # Might raise StopIteration
  File "/home/elias/hacking/scrapinghub/exporters/exporters/decompressors.py", line 24, in decompress
    for chunk in stream:
  File "/home/elias/hacking/scrapinghub/exporters/exporters/iterio.py", line 63, in next
    return self.next_chunk()
  File "/home/elias/hacking/scrapinghub/exporters/exporters/iterio.py", line 78, in next_chunk
    data = self._iterator.next()  # Might raise StopIteration
  File "/home/elias/hacking/scrapinghub/exporters/exporters/iterio.py", line 22, in iterate_chunks
    chunk = file.read(chunk_size)
  File "/usr/lib/python2.7/socket.py", line 380, in read
    data = self._sock.recv(left)
  File "/home/elias/hacking/scrapinghub/exporters/exporters/readers/s3_reader.py", line 18, in inner
    return func(*args)
  File "/usr/lib/python2.7/httplib.py", line 573, in read
    s = self.fp.read(amt)
  File "/usr/lib/python2.7/socket.py", line 380, in read
    data = self._sock.recv(left)
  File "/usr/lib/python2.7/ssl.py", line 341, in recv
    return self.read(buflen)
  File "/usr/lib/python2.7/ssl.py", line 260, in read
    return self._sslobj.read(len)
error: [Errno 104] Connection reset by peer

ERROR:export-pipeline: -- EXPORTMANAGER -- [Errno 104] Connection reset by peer
Traceback (most recent call last):
  File "bin/export.py", line 41, in <module>
    run(args)
  File "bin/export.py", line 36, in run
    exporter.export()
  File "/home/elias/hacking/scrapinghub/exporters/exporters/export_managers/base_exporter.py", line 178, in export
    raise e
socket.error: [Errno 104] Connection reset by peer
```